tests/integration/framework/test_fixtures.py

```diff
@@ -112,35 +112,39 @@ def test_dbfs_fixture(make_mounted_location):
 def test_remove_after_tag_jobs(ws, env_or_skip, make_job):
     new_job = make_job(spark_conf=_SPARK_CONF)
     created_job = ws.jobs.get(new_job.job_id)
-    assert "RemoveAfter" in created_job.settings.tags
+    job_tags = created_job.settings.tags
+    assert "RemoveAfter" in job_tags
 
-    purge_time = datetime.strptime(created_job.settings.tags.get("RemoveAfter"), "%Y%m%d%H")
-    assert purge_time - datetime.utcnow() < timedelta(hours=1, minutes=15)
+    purge_time = datetime.strptime(job_tags["RemoveAfter"], "%Y%m%d%H").replace(tzinfo=timezone.utc)
+    assert (purge_time - datetime.now(timezone.utc)) < (TEST_RESOURCE_PURGE_TIMEOUT + timedelta(hours=1))  # noqa: F405
```

Comment:
Out of curiosity, is there a difference between `datetime.utcnow()` and `datetime.now(timezone.utc))` ?

Reply:
There is: the former returns a naïve timestamp (implicitly placed in the "local" timezone, whatever that is) which is a bit of an oxymoron for UTC. It's also been [deprecated as of 3.12](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow).